### PR TITLE
Add overflow menu (`ActionMenu`) to focus zone in `ActionBar`

### PR DESCRIPTION
**Pull request by TylerJDev**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

PR adds the overflow menu (`ActionMenu`) to the focus trap within `ActionBar`. This is based off of the following issue: https://github.com/github/primer/issues/1131.

### Screenshots

<details>
<summary>Before PR, shows overflow menu being reachable only through tab</summary>

https://github.com/primer/view_components/assets/26746305/400516ef-a664-4bfc-aa17-1201cb0c71ca

Video shows arrow key navigation through the `ActionBar` component. The navigation is trapped within the menu items and does not reach the overflow menu. Video then shows usage of the `Tab` key, focusing directly onto the overflow menu.
</details>

<details>
<summary>Current PR, shows overflow menu being placed within focus trap</summary>

https://github.com/primer/view_components/assets/26746305/af57c86a-e6fe-457f-8930-96d6e10de6ff

Video shows arrow key navigation through the `ActionBar` component. The navigation is trapped within the component, and reaches the overflow menu without having to press `Tab` to focus onto it.
</details>

#### List the issues that this change affects.

https://github.com/github/primer/issues/1131

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?

     
PR adds `ActionMenu` to focus trap within `ActionBar`. Currently, the menu must be tabbed into separately rather than a part of the focus trap.
 
This also adds the ability in `focus_group` to have `tabindex` set back onto the invoker button of the popover menu. This fixes a bug in `ActionBar`, where if clicked outside of that menu whilst active, focus would be stuck on a `menuitem` and not the invoker.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/focus_group.ts [hunk 1]
```diff
@@ -111,7 +137,7 @@ export default class FocusGroupElement extends HTMLElement {
         let el: HTMLElement | null = focusEl
         do {
           el = el.closest(`[popover]:not(:popover-open)`)
-          if (el?.popover === 'auto') {
+          if (el?.popover === 'auto' && !['ArrowRight', 'ArrowLeft'].includes(event.key)) {
```
Comment (TylerJDev) [inline app/components/primer/focus_group.ts:140]: This prevents the left and right arrow keys from triggering a menu. Without, if you were to navigate the ActionBar with arrow keys, the menu would automatically open when either key is pressed.

**Review by camertron — APPROVED** [review]
Cool, thank you!

**Review by camertron — DISMISSED** [review]
Ah sorry. I approved but forgot I wanted to ask for some tests first 😄

**Review by jonrohan — APPROVED** [review]
Looks great thanks for the help!

**Pull request merged**